packages/core/src/browser/quick-open/quick-pick-service.ts

```diff
@@ -14,102 +14,28 @@
-    /**
-     * default: true
-     */
-    fuzzyMatchDescription?: boolean
-}
-
-@injectable()
-export class QuickPickService {
-
-    @inject(QuickOpenService)
-    protected readonly quickOpenService: QuickOpenService;
-
-    show(elements: string[], options?: QuickPickOptions): Promise<string | undefined>;
-    show<T>(elements: QuickPickItem<T>[], options?: QuickPickOptions): Promise<T | undefined>;
-    async show(elements: (string | QuickPickItem<Object>)[], options?: QuickPickOptions): Promise<Object | undefined> {
-        return new Promise<Object | undefined>(resolve => {
-            const items = this.toItems(elements, resolve);
-            if (items.length === 0) {
-                resolve(undefined);
-                return;
-            }
-            if (items.length === 1) {
-                items[0].run(QuickOpenMode.OPEN);
-                return;
-            }
-            this.quickOpenService.open({ onType: (_, acceptor) => acceptor(items) }, Object.assign({
-                onClose: () => resolve(undefined),
-                fuzzyMatchLabel: true,
-                fuzzyMatchDescription: true
-            }, options));
-        });
-    }
-    protected toItems(elements: (string | QuickPickItem<Object>)[], resolve: (element: Object) => void): QuickOpenItem[] {
-        const items: QuickOpenItem[] = [];
-        let groupLabel: string | undefined;
-        for (const element of elements) {
-            if (QuickPickSeparator.is(element)) {
-                groupLabel = element.label;
-            } else {
-                const options = this.toItemOptions(element, resolve);
-                if (groupLabel) {
-                    items.push(new QuickOpenGroupItem(Object.assign(options, { groupLabel, showBorder: true })));
-                    groupLabel = undefined;
-                } else {
-                    items.push(new QuickOpenItem(options));
-                }
-            }
-        }
-        return items;
-    }
-    protected toItemOptions(element: string | QuickPickValue<Object>, resolve: (element: Object) => void): QuickOpenItemOptions {
-        const label = typeof element === 'string' ? element : element.label;
-        const value = typeof element === 'string' ? element : element.value;
-        const description = typeof element === 'string' ? undefined : element.description;
-        const iconClass = typeof element === 'string' ? undefined : element.iconClass;
-        return {
-            label,
-            description,
-            iconClass,
-            run: mode => {
-                if (mode !== QuickOpenMode.OPEN) {
-                    return false;
-                }
-                resolve(value);
-                return true;
-            }
-        };
-    }
-
-}
+import * as common from '../../common/quick-pick-service';
+/**
+ * @deprecated import from `@theia/core/lib/common/quick-pick-service` instead
+ */
+export const QuickPickService = common.QuickPickService;
```

Comment:
why this service is becoming deprecated, what is the reason behind ?

Reply:
`QuickPickService` can be used on the backend as well know, because of it interface declarations we moved to common from browser. Developers should update imports as well.

Reply:
you're telling the "how", not "why" ?

Reply:
to be used in the java debug adapter contribution, please use find references on `QuickPickService`

Reply:
@akosyakov I see the choice you've used, not the reason why service should be callable from browser and node.

Reply:
@akosyakov The main point is that there was no discussion previously. it was just a comment on a PR. Also planned tasks on Theia from TypeFox are unclear. Also why the word `fast` is used ?
Also, the word `we` should be replaced by `TypeFox` probably ?

Reply:
@benoitf We all have many decisions to make on each PR. If there are some that you care about and want to discuss, please do so (like you did here).
It is pretty hard to know upfront what things others might want to discuss and we couldn't move anywhere if we need to have meetings for every decision. That is IMHO one important reason we are having PRs. Someone proposes something and people who care chime in.
But your comment sounds a bit alarming, so I think we should discuss this in tomorrow's meeting. I added it the agenda.